services/deploy/deploy.ps1

```diff
@@ -437,7 +442,7 @@ Function SelectAzureADTenantId() {
 Function ConnectToAzureADTenant() {
     if ($script:interactive) {
         # Interactive selecting tenant to connect to
-        if (!$script:tenantId) {
+        if (!$script:tenantId -or $script:type -ne "vm") {
```

Comment:
Could you change this for "vm" also?  And then just remove the SelectAzureADTenantId function.   Just use the tenant from the subscription.